src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java

```diff
@@ -787,10 +788,11 @@ public void onOutlawEnterTown(PlayerEnterTownEvent event) throws NotRegisteredEx
 		Town town = event.getEnteredtown();
 		
 		if (town.hasOutlaw(outlaw)) {
-			if (TownySettings.doTownsGetWarnedOnOutlaw()) {
+			if (TownySettings.doTownsGetWarnedOnOutlaw() && !TownyUniverse.getInstance().getPermissionSource().isTownyAdmin(event.getPlayer())) {
```

Comment:
This isTownyAdmin() test should happen earlier, before the outlaw variable is declared.

Reply:
Done